web/sfe/index.ts

```diff
@@ -0,0 +1,480 @@
+                new RedirectStage(this).render();
+                return;
+            case "ak-stage-autosubmit":
+                new AutosubmitStage(this).render();
+                return;
+            case "ak-stage-authenticator-validate":
+                new AuthenticatorValidateStage(this).render();
+                return;
+            default:
+                this.container.innerText = "Unsupported stage: " + this.challenge.component;
+                return;
+        }
+    }
+}
+
+class Stage {
+    constructor(public executor: SimpleFlowExecutor) {}
+
+    error(fieldName: string): { string: string; code: string }[] {
+        if (!this.executor.challenge.response_errors) {
+            return [];
+        }
+        return this.executor.challenge.response_errors[fieldName] || [];
+    }
+
+    renderInputError(fieldName: string) {
+        return `${this.error(fieldName)
+            .map((error) => {
+                return `<div class="invalid-feedback">
+                                ${error.string}
+                            </div>`;
+            })
+            .join("")}`;
+    }
+
+    html(html: string) {
+        this.executor.container.innerHTML = html;
+    }
+
+    render() {
+        throw new Error("Abstract method");
+    }
+}
+
+class IdentificationStage extends Stage {
+    render() {
+        this.html(`
+            <form id="ident-form">
+                <img class="mb-4 brand-icon" src="${ak().brand.branding_logo}" alt="">
+                <h1 class="h3 mb-3 fw-normal text-center">${this.executor.challenge.flow_info.title}</h1>
+                ${
+                    this.executor.challenge.application_pre
+                        ? `<p>
+                              Login to continue to ${this.executor.challenge.application_pre}.
+                          </p>`
+                        : ""
+                }
+                <div class="form-label-group my-3 has-validation">
+                    <input type="text" autofocus class="form-control" name="uid_field" placeholder="Email / Username">
+                </div>
+                ${
+                    this.executor.challenge.password_fields
+                        ? `<div class="form-label-group my-3 has-validation">
+                                <input type="password" class="form-control ${this.error("password").length > 0 ? "is-invalid" : ""}" name="password" placeholder="Password">
+                                ${this.renderInputError("password")}
+                        </div>`
+                        : ""
+                }
+                <button class="btn btn-primary w-100 py-2" type="submit">${this.executor.challenge.primary_action}</button>
+            </form>`);
+        $("#ident-form input[name=uid_field]").trigger("focus");
+        $("#ident-form").on("submit", (ev) => {
+            ev.preventDefault();
+            const data = new FormData(ev.target as HTMLFormElement);
+            this.executor.submit(data);
```

Comment:
My experience suggests that, rather than attempt to use these features in an IE-like environment, fall back on older-school tactics: 

```
        $("#ident-form").on("submit", (ev) => {
            const data = {
                uid_field: $('#ident-form input[name="uid_field"]').val(),
            };
            if (this.executor.challenge.password_fields) {
                data["password"] = $('#ident-form input[name="password"]').val();
            }
            this.executor.submit(data);
        });
```

Substituting this for the `uid_field`, and doing the equivalent for the `password` field, allowed me to log in.  The UserLibrary didn't resolve, but... 🤷‍♀️